### DISCUSSION TITLE
How do i set the Window.Owner property.

### DISCUSSION THREAD

**Discussion by SaverinOnRails** [AvaloniaUI/Avalonia#12845] [General]
I'm trying to set the window.Owner property on an avalonia window so i can prevent the window from showing up on window's task switcher. In WPF, from the context of a window you can just do this:
```
window.Owner = this
```
How is it done in avalonia?

**Comment by maxkatz6** [comment]
In Avalonia you set the Owner while calling Show or ShowDialog method. Owner is a single parameter there.

**Reply by SaverinOnRails** [reply]
Hi max, I'm trying to avoid using showDialog because if the 'dialog' window is open, users cannot interact with the main window as it is blocked off by the dialog. 


Edit
Lol I just re-read your message and realized I can also pass the owner to the show method. Thank you for your help.

**Comment by BreyerW** [comment]
Also if you just want to remove window from taskbar you can simply set `window.ShowInTaskbar=false`

EDIT: just noticed u mean switcher so probably not relevant but you can check if this works for you too

**Reply by SaverinOnRails** [reply]
Yes this only works for task switcher if the window is a dialog or has an owner
